src/math/log_10.c

```diff
@@ -0,0 +1,195 @@
+// SPDX-License-Identifier: BSD-3-Clause/
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+/* Include Files */
+#include <sof/math/log_10.h>
+
+/* Defines Constant*/
+#define BASE10LOG_DENOM_WORDLENGTH 0x1F
+#define BASE10LOG_WRAP_SCHAR_BITS 0xFF
+#define BASE10LOG_NUMER_WORDLENGTH 0x3
+#define BASE10LOG_UPPERBYTES 0xFFFFFF
+#define BASE10LOG_NUM_SHIFTBITS 0x1C
+/* Function Declarations */
+static int64_t division(int64_t numerator, int64_t denominator);
+
+/* Function Definitions */
+/*
+ * Arguments	: int64_t numerator
+ *		  int64_t denominator
+ * Return Type	: int64_t
+ */
+static int64_t division(int64_t numerator, int64_t denominator)
+{
+	int64_t quotient;
+	uint64_t absDenominator;
+	uint64_t absNumerator;
+	uint64_t tempAbsQuotient;
+	uint32_t quotientNeedsNegation;
+
+	absNumerator = (numerator < 0) ? ~numerator + 1 : numerator;
+	absDenominator = (denominator < 0) ? ~denominator + 1 : denominator;
+	quotientNeedsNegation = ((numerator < 0) != (denominator < 0));
+	tempAbsQuotient = absNumerator / absDenominator;
+	absNumerator %= absDenominator;
+	absNumerator <<= 1;
+	if (absNumerator >= absDenominator &&
+	    (!quotientNeedsNegation || absNumerator > absDenominator)) {
+		tempAbsQuotient++;
+	}
+
+	quotient = (quotientNeedsNegation) ? -(int64_t)tempAbsQuotient : (int64_t)tempAbsQuotient;
+	return quotient;
+}
+
+/**
+ * Base-10 logarithm  log10(x)
+ *
+ * loge = (u) computes the base-10 logarithm of
+ * u using lookup table.
+ * input u must be scalar/real number and positive
+ *
+ * +------------------+-----------------+--------+--------+
+ * | inpfxp	      |log10(returntype)| inpfxp |  loge  |
```

Comment:
s/loge/log10/
And I recommend to use UQm.n (https://en.wikipedia.org/wiki/Q_(number_format)) for unsigned type to avoid confusion.

Reply:
Thanks for link @johnylin76 , the version used in our documents is there the https://en.wikipedia.org/wiki/Q_(number_format)#AMD_version .

Reply:
included the suggestion in nomenclature